File: 백준/Silver/11047. 동전 0/동전 0.py
import sys
input = sys.stdin.readline
n, k = map(int, input().split())
coins = []
for _ in range(n):
    coin = int(input())
    coins.append(coin)
coins.reverse()
answer = 0
for coin in coins:
    answer += k // coin
    k = k % coin
    if k == 0:
        break
print(answer)  

File: 백준/Silver/11047. 동전 0/동전 0.swift
import Foundation

let input = readLine()!.split(separator: " ").map{Int(String($0))}

let num = input[0]
var arr: [Int] = []
var total: Int = input[1]!

var moneyStack = Stack()

var output:Int = 0

for _ in 1...num!{
    moneyStack.push(element: Int(readLine()!)!)
}

for _ in 1...num!{
    let money = moneyStack.pop()
    let count = (total / money)
    if(count >= 1){
        output = output + count
        total = total % money
    }
    if(total == 0){
        break
    }
}
print(output)

struct Stack{
    private var stack: [Int] = []
    private var isEmpty: Bool {
        return stack.count == 0
    }
    public mutating func push(element: Int){
        stack.append(element)
    }
    public mutating func pop() -> Int {
        return stack.removeLast()
    }
}
